"charged" state sometimes does not display when battery at 100% and plugged in

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Actual behaviour:
Sometimes, when MacOS reports fully charged battery and power is plugged in, the default formatting is applied (no icon and default foreground colour):
![Screenshot 2023-04-19 at 1 07 11 pm](https://user-images.githubusercontent.com/46676673/232956701-b26451a6-ed45-4bba-adcd-7fcaabf15921.png)
When in this state, if the charging cable is unplugged and plugged back in, then the expected behaviour occurs again.

Expected behaviour:
When the battery is at 100%, the "charged" icon and foreground format should be displayed:
![Screenshot 2023-04-19 at 1 05 34 pm](https://user-images.githubusercontent.com/46676673/232956445-6a0e5d6a-787e-4eda-8ca5-0875242f3052.png)

### Theme

Slightly edited [bubblesextra](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/bubblesextra.omp.json). Different colours and icons used.

### What OS are you seeing the problem on?

macOS

### Which shell are you using?

zsh

### Log output

```Shell
Version: 14.30.0

Shell: zsh (5.9)

Segments:

ConsoleTitle(true)                         -   0 ms - zsh in themes
python(false)                              -   0 ms -
go(false)                                  -   0 ms -
node(false)                                -   0 ms -
ruby(false)                                -   0 ms -
java(false)                                -   0 ms -
julia(false)                               -   0 ms -
php(false)                                 -   0 ms -
git(true)                                  -  21 ms - main ≢
path(true)                                 -   0 ms -   ~/.oh-my-posh/themes
battery(true)                              -   8 ms -  ⇣ 100
executiontime(false)                       -   0 ms -
text(true)                                 -   0 ms - tom
exit(true)                                 -   0 ms - ❯

Run duration: 33.300792ms

Cache path: /Users/tom

Config path: /Users/tom/.oh-my-posh/themes/tom.omp.json

Logs:

[DEBUG] 13:17:02.071 shell.go:Getenv:398 → iTerm.app
[TRACE] 13:17:02.071 shell.go:Getenv(TERM_PROGRAM) - 128.208µs
[DEBUG] 13:17:02.071 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.071 shell.go:Getenv(XDG_CACHE_HOME) - 2.125µs
[TRACE] 13:17:02.071 shell_unix.go:CachePath() - 36.958µs
[DEBUG] 13:17:02.071 shell.go:Getenv:398 → /Users/tom/.oh-my-posh/themes/tom.omp.json
[TRACE] 13:17:02.071 shell.go:Getenv(POSH_THEME) - 3.75µs
[DEBUG] 13:17:02.071 shell_unix.go:Platform:71 → darwin
[TRACE] 13:17:02.071 shell.go:resolveConfigPath() - 31.917µs
[TRACE] 13:17:02.071 shell.go:Init() - 430.458µs
[TRACE] 13:17:02.071 shell.go:Flags() - 42ns
[TRACE] 13:17:02.072 config.go:loadConfig() - 711.875µs
[TRACE] 13:17:02.072 shell.go:Flags() - 42ns
[DEBUG] 13:17:02.072 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.072 shell.go:Getenv(OMP_CACHE_DISABLED) - 2.666µs
[DEBUG] 13:17:02.072 shell.go:Shell:645 → process name: zsh
[TRACE] 13:17:02.072 shell.go:Shell() - 119.417µs
[DEBUG] 13:17:02.072 shell.go:Getenv:398 → 5.9
[TRACE] 13:17:02.072 shell.go:Getenv(POSH_SHELL_VERSION) - 2.209µs
[DEBUG] 13:17:02.072 engine.go:PrintDebug:266 → Segment: Title
[TRACE] 13:17:02.072 shell_unix.go:Root() - 2.25µs
[TRACE] 13:17:02.072 shell.go:Shell() - 41ns
[TRACE] 13:17:02.072 shell.go:ErrorCode() - 209ns
[DEBUG] 13:17:02.072 shell_unix.go:IsWsl:35 → false
[TRACE] 13:17:02.072 shell_unix.go:IsWsl() - 20.292µs
[TRACE] 13:17:02.072 shell.go:GOOS() - 125ns
[DEBUG] 13:17:02.072 shell.go:Pwd:428 → /Users/tom/.oh-my-posh/themes
[TRACE] 13:17:02.072 shell.go:Pwd() - 35.333µs
[TRACE] 13:17:02.072 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 83ns
[DEBUG] 13:17:02.072 shell.go:User:549 → tom
[TRACE] 13:17:02.072 shell.go:User() - 2µs
[DEBUG] 13:17:02.072 shell.go:Host:561 → ARY4PJGC4VV
[TRACE] 13:17:02.072 shell.go:Host() - 14.084µs
[TRACE] 13:17:02.072 shell.go:GOOS() - 42ns
[DEBUG] 13:17:02.072 shell.go:Getenv:398 → 1
[TRACE] 13:17:02.072 shell.go:Getenv(SHLVL) - 1.916µs
[TRACE] 13:17:02.072 shell.go:TemplateCache() - 145.458µs
[TRACE] 13:17:02.072 shell.go:Flags() - 42ns
[DEBUG] 13:17:02.072 block.go:Debug:230 → Segment: python
[DEBUG] 13:17:02.072 properties.go:GetString:28 → environment
[TRACE] 13:17:02.072 shell.go:Pwd() - 83ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 83ns
[DEBUG] 13:17:02.072 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 13:17:02.072 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv env venv-win pyenv-win]
[DEBUG] 13:17:02.072 properties.go:GetBool:22 → home_enabled: true
[TRACE] 13:17:02.072 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.072 properties.go:GetBool:22 → fetch_virtual_env: true
[DEBUG] 13:17:02.072 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.072 shell.go:Getenv(VIRTUAL_ENV) - 1.625µs
[DEBUG] 13:17:02.072 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.072 shell.go:Getenv(CONDA_ENV_PATH) - 1.625µs
[DEBUG] 13:17:02.072 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.072 shell.go:Getenv(CONDA_DEFAULT_ENV) - 1.5µs
[DEBUG] 13:17:02.072 block.go:Debug:230 → Segment: go
[TRACE] 13:17:02.072 shell.go:Pwd() - 84ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 42ns
[DEBUG] 13:17:02.072 properties.go:GetStringArray:52 → extensions: [*.go go.mod]
[DEBUG] 13:17:02.072 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:17:02.072 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:17:02.072 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.072 properties.go:GetString:28 → files
[TRACE] 13:17:02.072 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.072 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.072 shell.go:HasFiles(*.go) - 105.667µs
[TRACE] 13:17:02.072 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.072 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.072 shell.go:HasFiles(go.mod) - 9.791µs
[DEBUG] 13:17:02.072 block.go:Debug:230 → Segment: node
[TRACE] 13:17:02.072 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.072 shell.go:GOOS() - 41ns
[DEBUG] 13:17:02.072 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .npmrc .vue]
[DEBUG] 13:17:02.072 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:17:02.072 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:17:02.072 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.072 properties.go:GetString:28 → files
[DEBUG] 13:17:02.072 properties.go:GetBool:22 → fetch_package_manager: false
[TRACE] 13:17:02.072 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.072 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.072 shell.go:HasFiles(*.js) - 32.042µs
[TRACE] 13:17:02.072 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.ts) - 27.584µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(package.json) - 7.083µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(.nvmrc) - 6.917µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(pnpm-workspace.yaml) - 7.916µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(.pnpmfile.cjs) - 6.458µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(.npmrc) - 6.541µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(.vue) - 11.083µs
[DEBUG] 13:17:02.073 block.go:Debug:230 → Segment: ruby
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 42ns
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → extensions: [*.rb Rakefile Gemfile]
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:17:02.073 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 properties.go:GetString:28 → files
[TRACE] 13:17:02.073 shell.go:Pwd() - 0s
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.rb) - 27.709µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(Rakefile) - 6.292µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(Gemfile) - 6.667µs
[DEBUG] 13:17:02.073 block.go:Debug:230 → Segment: java
[DEBUG] 13:17:02.073 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.073 shell.go:Getenv(JAVA_HOME) - 1.5µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 41ns
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → extensions: [pom.xml build.gradle.kts build.sbt .java-version .deps.edn project.clj build.boot *.java *.class *.gradle *.jar *.clj *.cljc]
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:17:02.073 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 properties.go:GetString:28 → files
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(pom.xml) - 6.75µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(build.gradle.kts) - 6.541µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(build.sbt) - 7.042µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(.java-version) - 10.084µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(.deps.edn) - 6.375µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(project.clj) - 6.542µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(build.boot) - 5.417µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.java) - 26.375µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.class) - 25.958µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.gradle) - 24.75µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.jar) - 23.833µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 83ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.clj) - 22.709µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.cljc) - 22.5µs
[DEBUG] 13:17:02.073 block.go:Debug:230 → Segment: julia
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 84ns
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → extensions: [*.jl]
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:17:02.073 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 properties.go:GetString:28 → files
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.jl) - 21.709µs
[DEBUG] 13:17:02.073 block.go:Debug:230 → Segment: php
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 41ns
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → extensions: [*.php composer.json composer.lock .php-version blade.php]
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:17:02.073 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 properties.go:GetString:28 → files
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(*.php) - 21.25µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 83ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(composer.json) - 7.709µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(composer.lock) - 7.083µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(.php-version) - 6.708µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[DEBUG] 13:17:02.073 shell.go:HasFiles:450 → false
[TRACE] 13:17:02.073 shell.go:HasFiles(blade.php) - 49.667µs
[DEBUG] 13:17:02.073 block.go:Debug:230 → Segment: git
[TRACE] 13:17:02.073 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 41ns
[DEBUG] 13:17:02.073 shell_unix.go:IsWsl:35 → false
[TRACE] 13:17:02.073 shell_unix.go:IsWsl() - 2.208µs
[TRACE] 13:17:02.073 shell_unix.go:IsWsl2() - 4µs
[TRACE] 13:17:02.073 shell.go:GOOS() - 41ns
[DEBUG] 13:17:02.073 shell.go:CommandPath:600 → /usr/bin/git
[TRACE] 13:17:02.073 shell.go:CommandPath(git) - 176.083µs
[TRACE] 13:17:02.073 shell.go:HasCommand(git) - 178.125µs
[TRACE] 13:17:02.073 shell.go:Pwd() - 41ns
[TRACE] 13:17:02.073 shell.go:HasParentFilePath(.git) - 18.917µs
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → exclude_folders: []
[TRACE] 13:17:02.073 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 84ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 0s
[TRACE] 13:17:02.073 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.073 shell.go:GOOS() - 0s
[DEBUG] 13:17:02.073 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.073 shell.go:Getenv(POSH_GIT_STATUS) - 1.75µs
[DEBUG] 13:17:02.073 properties.go:GetBool:22 → fetch_status: true
[DEBUG] 13:17:02.073 properties.go:GetStringArray:52 → ignore_status: []
[TRACE] 13:17:02.073 shell.go:GOOS() - 0s
[DEBUG] 13:17:02.073 properties.go:GetKeyValueMap:46 → untracked_modes: map[]
[DEBUG] 13:17:02.073 properties.go:GetKeyValueMap:46 → ignore_submodules: map[]
[DEBUG] 13:17:02.086 shell.go:RunCommand:579 ↓
    # branch.oid 61dac295e909bfb116709c3b68edd81c2f22209c
    # branch.head main
[TRACE] 13:17:02.086 shell.go:RunCommand(git -C /Users/tom/.oh-my-posh/themes --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 12.419416ms
[DEBUG] 13:17:02.086 properties.go:GetString:28 → 
[DEBUG] 13:17:02.086 properties.go:GetInt:40 → branch_max_length: 25
[DEBUG] 13:17:02.086 shell.go:HasFolder:493 → false
[TRACE] 13:17:02.086 shell.go:HasFolder(/Users/tom/.oh-my-posh/themes/.git/rebase-merge) - 13.625µs
[DEBUG] 13:17:02.086 shell.go:HasFolder:493 → false
[TRACE] 13:17:02.086 shell.go:HasFolder(/Users/tom/.oh-my-posh/themes/.git/rebase-apply) - 8.209µs
[DEBUG] 13:17:02.086 properties.go:GetString:28 → 
[DEBUG] 13:17:02.086 shell.go:HasFilesInDir:464 → false
[TRACE] 13:17:02.086 shell.go:HasFilesInDir(MERGE_MSG) - 5.125µs
[DEBUG] 13:17:02.086 shell.go:HasFilesInDir:464 → false
[TRACE] 13:17:02.086 shell.go:HasFilesInDir(CHERRY_PICK_HEAD) - 10.833µs
[DEBUG] 13:17:02.086 shell.go:HasFilesInDir:464 → false
[TRACE] 13:17:02.086 shell.go:HasFilesInDir(REVERT_HEAD) - 3.292µs
[DEBUG] 13:17:02.086 shell.go:HasFilesInDir:464 → false
[TRACE] 13:17:02.086 shell.go:HasFilesInDir(sequencer/todo) - 3.125µs
[DEBUG] 13:17:02.086 properties.go:GetString:28 → ≢
[DEBUG] 13:17:02.086 properties.go:GetBool:22 → fetch_upstream_icon: true
[ERROR] 13:17:02.093 shell.go:RunCommand:577 → exit status 2
[DEBUG] 13:17:02.093 shell.go:RunCommand:579 ↓
    error: No such remote 'origin'

[TRACE] 13:17:02.093 shell.go:RunCommand(git -C /Users/tom/.oh-my-posh/themes --no-optional-locks -c core.quotepath=false -c color.status=false remote get-url origin) - 7.562917ms
[TRACE] 13:17:02.093 shell.go:TemplateCache() - 42ns
[TRACE] 13:17:02.094 shell.go:TemplateCache() - 42ns
[ERROR] 13:17:02.094 shell.go:FileContent:519 → open /Users/tom/.oh-my-posh/themes/.git/logs/refs/stash: no such file or directory
[TRACE] 13:17:02.094 shell.go:FileContent(/Users/tom/.oh-my-posh/themes/.git/logs/refs/stash) - 42.75µs
[TRACE] 13:17:02.094 shell.go:Shell() - 125ns
[TRACE] 13:17:02.094 shell.go:TemplateCache() - 0s
[TRACE] 13:17:02.094 shell.go:TemplateCache() - 125ns
[TRACE] 13:17:02.095 shell.go:TemplateCache() - 125ns
[TRACE] 13:17:02.095 shell.go:TemplateCache() - 42ns
[DEBUG] 13:17:02.095 block.go:Debug:230 → Segment: path
[TRACE] 13:17:02.095 shell.go:Pwd() - 125ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.095 shell.go:Pwd() - 84ns
[TRACE] 13:17:02.095 shell.go:Shell() - 42ns
[TRACE] 13:17:02.095 shell.go:Shell() - 42ns
[DEBUG] 13:17:02.095 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 13:17:02.095 properties.go:GetString:28 → 
[DEBUG] 13:17:02.095 properties.go:GetString:28 → 
[TRACE] 13:17:02.095 shell.go:GOOS() - 41ns
[DEBUG] 13:17:02.095 properties.go:GetString:28 → ~
[DEBUG] 13:17:02.095 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 0s
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 41ns
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 41ns
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 41ns
[DEBUG] 13:17:02.095 properties.go:GetString:28 → full
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 0s
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 0s
[DEBUG] 13:17:02.095 properties.go:GetString:28 → NO DATA
[TRACE] 13:17:02.095 shell.go:PathSeparator() - 41ns
[DEBUG] 13:17:02.095 properties.go:GetString:28 → /
[TRACE] 13:17:02.095 shell.go:Pwd() - 42ns
[DEBUG] 13:17:02.095 shell_unix.go:IsWsl:35 → false
[TRACE] 13:17:02.095 shell_unix.go:IsWsl() - 2.375µs
[TRACE] 13:17:02.095 shell.go:StackCount() - 41ns
[TRACE] 13:17:02.095 shell_unix.go:DirIsWritable(/Users/tom/.oh-my-posh/themes) - 8.75µs
[TRACE] 13:17:02.095 shell.go:TemplateCache() - 0s
[TRACE] 13:17:02.095 shell.go:TemplateCache() - 42ns
[TRACE] 13:17:02.095 shell.go:Shell() - 83ns
[DEBUG] 13:17:02.095 block.go:Debug:230 → Segment: battery
[TRACE] 13:17:02.095 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.095 shell.go:GOOS() - 41ns
[DEBUG] 13:17:02.095 shell_unix.go:IsWsl:35 → false
[TRACE] 13:17:02.095 shell_unix.go:IsWsl() - 1.792µs
[DEBUG] 13:17:02.103 shell.go:RunCommand:579 ↓
    Now drawing from 'Battery Power'
     -InternalBattery-0 (id=6094947)	100%; discharging; 7:29 remaining present: true
[TRACE] 13:17:02.103 shell.go:RunCommand(pmset -g batt) - 7.543417ms
[TRACE] 13:17:02.103 shell_darwin.go:BatteryState() - 7.585291ms
[DEBUG] 13:17:02.103 properties.go:GetString:28 → ⇣
[TRACE] 13:17:02.103 shell.go:TemplateCache() - 42ns
[TRACE] 13:17:02.103 shell.go:TemplateCache() - 41ns
[TRACE] 13:17:02.103 shell.go:Shell() - 42ns
[TRACE] 13:17:02.103 shell.go:TemplateCache() - 42ns
[TRACE] 13:17:02.103 shell.go:TemplateCache() - 42ns
[TRACE] 13:17:02.103 shell.go:Flags() - 42ns
[DEBUG] 13:17:02.103 block.go:Debug:230 → Segment: executiontime
[TRACE] 13:17:02.103 shell.go:Pwd() - 167ns
[TRACE] 13:17:02.103 shell.go:GOOS() - 42ns
[DEBUG] 13:17:02.103 properties.go:GetBool:22 → always_enabled: false
[TRACE] 13:17:02.103 shell.go:ExecutionTime() - 84ns
[DEBUG] 13:17:02.103 properties.go:GetFloat64:34 → threshold: 250.000000
[DEBUG] 13:17:02.103 block.go:Debug:230 → Segment: text
[TRACE] 13:17:02.103 shell.go:Pwd() - 41ns
[TRACE] 13:17:02.103 shell.go:GOOS() - 42ns
[TRACE] 13:17:02.103 shell.go:TemplateCache() - 41ns
[TRACE] 13:17:02.104 shell.go:TemplateCache() - 42ns
[TRACE] 13:17:02.104 shell.go:Shell() - 83ns
[DEBUG] 13:17:02.104 block.go:Debug:230 → Segment: exit
[TRACE] 13:17:02.104 shell.go:Pwd() - 42ns
[TRACE] 13:17:02.104 shell.go:GOOS() - 0s
[TRACE] 13:17:02.104 shell.go:ErrorCode() - 83ns
[DEBUG] 13:17:02.104 properties.go:GetBool:22 → always_enabled: true
[TRACE] 13:17:02.104 shell.go:TemplateCache() - 41ns
[TRACE] 13:17:02.104 shell.go:Shell() - 42ns
[TRACE] 13:17:02.104 shell.go:TemplateCache() - 41ns
[DEBUG] 13:17:02.104 shell.go:Getenv:398 → NO DATA
[TRACE] 13:17:02.104 shell.go:Getenv(XDG_CACHE_HOME) - 1.875µs
[TRACE] 13:17:02.104 shell_unix.go:CachePath() - 19.25µs
[TRACE] 13:17:02.104 shell.go:Flags() - 42ns
```

--- a/src/platform/battery/battery_darwin.go
+++ b/src/platform/battery/battery_darwin.go
@@ -34,11 +34,22 @@ func parseBatteryOutput(output string) (*Info, error) {
 		msg := "Unable to find battery state based on output"
 		return nil, errors.New(msg)
 	}
+
 	var percentage int
 	var err error
 	if percentage, err = strconv.Atoi(matches["PERCENTAGE"]); err != nil {
 		return nil, errors.New("Unable to parse battery percentage")
 	}
+
+	// sometimes it reports discharging when at 100, so let's force it to Full
+	// https://github.com/JanDeDobbeleer/oh-my-posh/issues/3729
+	if percentage == 100 {
+		return &Info{
+			Percentage: percentage,
+			State:      Full,
+		}, nil
+	}
+
 	return &Info{
 		Percentage: percentage,
 		State:      mapMostLogicalState(matches["STATE"]),

--- a/src/platform/battery/battery_darwin_test.go
+++ b/src/platform/battery/battery_darwin_test.go
@@ -39,9 +39,9 @@ func TestParseBatteryOutput(t *testing.T) {
 			ExpectedPercentage: 100,
 		},
 		{
-			Case:               "discharging",
+			Case:               "discharging, but not",
 			Output:             "100%; discharging;",
-			ExpectedState:      Discharging,
+			ExpectedState:      Full,
 			ExpectedPercentage: 100,
 		},
 	}

